src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -112,8 +120,25 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
         artworkLocation: values.artworkLocation,
       })
 
-      // Remove photos marked for deletion
+      // Store images locally
+      if (
+        artworkId &&
+        artworkFormImagesRef.current &&
+        values.newPhotos.length
+      ) {
+        await artworkFormImagesRef.current?.saveImagesToLocalStorage(artworkId)
+        await setLocalImagesStoreLastUpdatedAt(
+          IMAGES_LOCAL_STORE_LAST_UPDATED_AT
+        )
+      }
 
+      // If the user is editing, we need to wait a bit until gemini starts
+      // returning the new processing images array
+      if (isEditing && values.newPhotos.length) {
+        await wait(2000)
+      }
```

Comment:
**question:** Do we still need to wait or can we remove this logic?

Reply:
yes- the issue is that so far I am not adding support to showing local images inside the carousel since that would take forever to implement, I am therefore here waiting until Gemini at least creates the first version of the image that is not available for deep zoom but to show the user it's there